src/extensions/default/HealthData/HealthDataManager.js

```diff
@@ -336,6 +330,17 @@ define(function (require, exports, module) {
         return result.promise();
     }
 
+    /**
+     * This function is auto called after 24 hours to empty the map
+     * Map is used to make sure that we send an event only once per 24 hours
+     **/
+
+    function emptyAnalyticsMap() {
+        HealthLogger.analyticsEventMap.clear();
```

Comment:
Shouldn't we just clear it once per day. Once cleared we shouldn't clear it for any other sessions as well.
Same for sending the data.
This wouldn't work across sessions right?